### DISCUSSION TITLE
Grok implementation

### DISCUSSION THREAD

**Discussion by noahfraiture** [open-webui/open-webui#6714] [General]
Hello,
The api of Grok from xAI has been released with 25€ free/month for the next two month. It's extremely similar to openai API as it requires a single change of the API key and the base URL to works.
I will become a major actor in AI in the next few months and it would be great to have access to their API in this platform just like we have access to OpenAI api

**Comment by tjbck** [comment]
This would be great as a Function, contribution welcome here! (e.g. https://openwebui.com/f/hub/openai_proxy_pipe)

**Reply by noahfraiture** [reply]
Ok thank you, will do when I have time ! It should be done in a few days if I don't encounter a personal problem or if I have difficulties to make the contribution

**Comment by spammenotinoz** [comment]
As it is OpenAI compatible It is natively supported already, just add into Connections. 
![image](https://github.com/user-attachments/assets/a7d8ce8d-fab8-456f-9d80-88fb85e279ea)

Grok is a really excellent and fast model
